core/src/test/java/org/infinispan/persistence/sifs/SoftIndexFileStoreRestartTest.java

```diff
@@ -238,4 +238,20 @@ public void testRestartWithEntryUpdatedMultipleTimes(boolean leafOrNode) throws
       // Other tests need a cache manager still
       createCacheManagers();
    }
+
+   @Test(dataProvider = "booleans")
+   public void testRestartSameKey(boolean deleteIndex) throws Throwable {
+      int size = 20;
+      for (int i = 0; i < size; i++) {
+         cache(0, cacheName).put("same-key", "value-" + i);
+      }
+
+      killMember(0, cacheName);
+
+      if (deleteIndex) {
+         // Delete the index which should force it to rebuild
+         Util.recursiveFileRemove(Paths.get(tmpDirectory, "index"));
+      }
+      createCacheManagers();
```

Comment:
could you check the key's value after restart?

Reply:
That is already done in other tests. If you undo the other fix the teardown phase of the test will hang :(

Reply:
I can rename the test though to better show what it is actually testing ;)